File: backend/app/api/summaries.py
"""纪要 API 路由"""

import uuid

from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy import select
from sqlalchemy.ext.asyncio import AsyncSession

from app.api.deps import get_db
from app.schemas.summary import (
    SummaryResponse,
    SummaryUpdate,
    ActionItemResponse,
    ActionItemUpdate,
    RiskResponse,
    MeetingSummaryResponse,
    SummaryListItemResponse,
)
from app.services.summary_service import summary_service
from app.services.meeting_service import meeting_service

# 全局纪要路由（不带 meeting_id 前缀）
global_router = APIRouter(prefix="/summaries", tags=["纪要管理"])

# 会议级纪要路由
router = APIRouter(prefix="/meetings/{meeting_id}", tags=["纪要管理"])


@global_router.get("", response_model=list[SummaryListItemResponse])
async def list_summaries(
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    db: AsyncSession = Depends(get_db),
):
    """获取所有纪要列表"""
    skip = (page - 1) * page_size
    rows, _ = await summary_service.list_summaries(db, skip=skip, limit=page_size)
    return [
        SummaryListItemResponse(
            id=str(s.id),
            meeting_id=str(s.meeting_id),
            meeting_title=m.title,
            content=s.content[:200] if s.content else "",
            status=s.status,
            created_at=s.created_at,
        )
        for s, m in rows
    ]


@router.post("/summarize", response_model=SummaryResponse, status_code=200)
async def generate_summary(meeting_id: uuid.UUID, db: AsyncSession = Depends(get_db)):
    """触发 Multi-Agent 生成纪要"""
    meeting = await meeting_service.get_meeting(db, meeting_id)
    if not meeting:
        raise HTTPException(status_code=404, detail="会议不存在")

    summary = await summary_service.generate_summary(db, meeting_id)
    if not summary:
        raise HTTPException(status_code=400, detail="无会议原文，无法生成纪要")
    return summary


@router.get("/summary", response_model=MeetingSummaryResponse)
async def get_meeting_summary(meeting_id: uuid.UUID, db: AsyncSession = Depends(get_db)):
    """获取会议纪要综合数据（纪要 + 行动项）。"""
    meeting = await meeting_service.get_meeting(db, meeting_id)
    if not meeting:
        raise HTTPException(status_code=404, detail="会议不存在")

    summary = await summary_service.get_summary(db, meeting_id)
    action_items = await summary_service.get_action_items(db, meeting_id)
    return MeetingSummaryResponse(
        summary=summary,
        action_items=action_items,
        risks=[],
    )


@router.get("/summary/detail", response_model=SummaryResponse)
async def get_summary_detail(meeting_id: uuid.UUID, db: AsyncSession = Depends(get_db)):
    """获取纪要详情"""
    summary = await summary_service.get_summary(db, meeting_id)
    if not summary:
        raise HTTPException(status_code=404, detail="纪要尚未生成")
    return summary


@router.patch("/summary", response_model=SummaryResponse)
async def update_summary(
    meeting_id: uuid.UUID,
    data: SummaryUpdate,
    db: AsyncSession = Depends(get_db),
):
    """保存人工校对后的组会纪要。"""
    meeting = await meeting_service.get_meeting(db, meeting_id)
    if not meeting:
        raise HTTPException(status_code=404, detail="会议不存在")

    summary = await summary_service.update_summary(db, meeting, data.content)
    if not summary:
        raise HTTPException(status_code=404, detail="纪要尚未生成")
    return summary


@router.get("/action-items", response_model=list[ActionItemResponse])
async def get_action_items(meeting_id: uuid.UUID, db: AsyncSession = Depends(get_db)):
    """获取行动项列表"""
    return await summary_service.get_action_items(db, meeting_id)


@router.patch("/action-items/{item_id}", response_model=ActionItemResponse)
async def update_action_item(
    meeting_id: uuid.UUID,
    item_id: uuid.UUID,
    data: ActionItemUpdate,
    db: AsyncSession = Depends(get_db),
):
    """更新行动项状态"""
    from app.models.action_item import ActionItem

    result = await db.execute(
        select(ActionItem).where(
            ActionItem.id == item_id,
            ActionItem.meeting_id == meeting_id,
        )
    )
    item = result.scalar_one_or_none()
    if not item:
        raise HTTPException(status_code=404, detail="行动项不存在")

    update_data = data.model_dump(exclude_unset=True)
    for key, value in update_data.items():
        setattr(item, key, value)

    await db.flush()
    await db.refresh(item)
    return item


@router.get("/risks", response_model=list[RiskResponse])
async def get_risks(meeting_id: uuid.UUID, db: AsyncSession = Depends(get_db)):
    """获取风险列表"""
    return await summary_service.get_risks(db, meeting_id)

File: frontend/src/api/summaries.ts
import { apiClient } from './client'
import type {
  Summary,
  ActionItem,
  Risk,
  MeetingSummary,
  SummaryListItem,
} from '@/types'
import {
  IS_DEMO_MODE,
  demoGenerateSummary,
  demoGetActionItems,
  demoGetMeetingSummary,
  demoGetRisks,
  demoGetSummaryDetail,
  demoListSummaries,
  demoUpdateSummary,
  demoUpdateActionItem,
} from '@/lib/demo-data'

// 获取所有纪要列表
export async function listSummaries(
  page = 1,
  pageSize = 20,
): Promise<SummaryListItem[]> {
  if (IS_DEMO_MODE) return demoListSummaries(page, pageSize)
  return apiClient
    .get('summaries', {
      searchParams: { page, page_size: pageSize },
    })
    .json()
}

// 生成纪要（触发 Multi-Agent）
export async function generateSummary(meetingId: string): Promise<Summary> {
  if (IS_DEMO_MODE) return demoGenerateSummary(meetingId)
  return apiClient.post(`meetings/${meetingId}/summarize`).json()
}

// 获取会议纪要综合数据（纪要 + 行动项 + 风险）
export async function getMeetingSummary(meetingId: string): Promise<MeetingSummary> {
  if (IS_DEMO_MODE) return demoGetMeetingSummary(meetingId)
  return apiClient.get(`meetings/${meetingId}/summary`).json()
}

// 获取纪要详情
export async function getSummaryDetail(meetingId: string): Promise<Summary> {
  if (IS_DEMO_MODE) return demoGetSummaryDetail(meetingId)
  return apiClient.get(`meetings/${meetingId}/summary/detail`).json()
}

export async function updateSummary(meetingId: string, content: string): Promise<Summary> {
  if (IS_DEMO_MODE) return demoUpdateSummary(meetingId, content)
  return apiClient.patch(`meetings/${meetingId}/summary`, { json: { content } }).json()
}

// 获取行动项列表
export async function getActionItems(meetingId: string): Promise<ActionItem[]> {
  if (IS_DEMO_MODE) return demoGetActionItems(meetingId)
  return apiClient.get(`meetings/${meetingId}/action-items`).json()
}

// 更新行动项
export async function updateActionItem(
  meetingId: string,
  itemId: string,
  data: { status?: string; priority?: string; assignee?: string },
): Promise<ActionItem> {
  if (IS_DEMO_MODE) return demoUpdateActionItem(meetingId, itemId, data as Partial<ActionItem>)
  return apiClient
    .patch(`meetings/${meetingId}/action-items/${itemId}`, { json: data })
    .json()
}

// 获取风险列表
export async function getRisks(meetingId: string): Promise<Risk[]> {
  if (IS_DEMO_MODE) return demoGetRisks(meetingId)
  return apiClient.get(`meetings/${meetingId}/risks`).json()
}
